ql/src/java/org/apache/hadoop/hive/ql/optimizer/StatsOptimizer.java

```diff
@@ -397,6 +397,11 @@ else if (getGbyKeyType(cgbyOp) == GbyKeyType.CONSTANT && rowCnt == 0) {
           if (udaf instanceof GenericUDAFSum) {
             // long/double/decimal
             ExprNodeDesc desc = aggr.getParameters().get(0);
+            // add null for SUM(1), when the table is empty. Without this, category = LONG, and the result is 0
+            // instead of NULL.
+            if (desc instanceof ExprNodeConstantDesc && rowCnt == 0) {
+              oneRow.add(null);
+            }
```

Comment:
down below it will add a second value to the row (`oneRow`) ; try a test with multiple sum-s like 2